src/ast/mod.rs

```diff
@@ -3058,6 +3058,33 @@ pub enum Statement {
         begin: bool,
         transaction: Option<BeginTransactionKind>,
         modifier: Option<TransactionModifier>,
+        /// List of statements belonging to the `BEGIN` block.
+        /// Example:
+        /// ```sql
+        /// BEGIN
+        ///     SELECT 1;
+        ///     SELECT 2;
+        /// END;
+        /// ```
+        statements: Vec<Statement>,
+        /// TRUE if the statement has a
+        /// `EXCEPTION WHEN ERROR THEN` clause
+        /// Example:
+        /// ```sql
+        /// BEGIN
+        ///     SELECT 1;
+        ///     SELECT 2;
+        /// EXCEPTION WHEN ERROR THEN
+        ///     SELECT 3;
+        /// END;
+        /// ```
+        /// <https://cloud.google.com/bigquery/docs/reference/standard-sql/procedural-language#beginexceptionend>
+        has_exception_when_clause: bool,
```

Comment:
I wondered if this field redundant with when `exception_statements` is empty? However, given the test cases below, it seems it is possible to have a `EXCEPTION WHEN ERROR THEN END`  (without any statements) so distinguishing is necessary

BTW another more Rusty way might be to represent this as 
```rust
        exception_statements: Option<Vec<Statement>>,
```

I don't feel strongly about this however

Reply:
Yeah I think that makes sense, I initially figured to use an explicit flag in case it was possible to have exception statements with a different syntax but probably no need to do that unless necessary, will update to use an option!

Reply:
I think using the somewhat akward `Option<Vec<..>>` would allow the different syntax.

I can help if that would be good

Reply:
Done! updated to use an `Option`